components/InventoryItemListItem/index.tsx

```diff
@@ -0,0 +1,157 @@
+import { TableRow, TableCell, Chip } from '@mui/material'
+import { InventoryItem } from 'utils/types'
+import WarningIcon from '@mui/icons-material/Warning'
+import { MoreVert } from '@mui/icons-material'
+import colors from 'utils/colors'
+
+interface InventoryItemListItemProps {
+  inventoryItem: InventoryItem
+}
+
+export default function InventoryItemListItem({
+  inventoryItem,
+}: InventoryItemListItemProps) {
+  const narrowCategory = (inventoryItem: InventoryItem) => {
+    // needed to deal with the possible union types defined in the database schema
+    if (
+      typeof inventoryItem.itemDefinition === 'string' ||
+      !inventoryItem.itemDefinition.category
+    ) {
+      return ' '
+    }
+
+    if (typeof inventoryItem.itemDefinition.category === 'string') {
+      return inventoryItem.itemDefinition.category
+    }
+
+    return inventoryItem.itemDefinition.category.name
+  }
+
+  const renderWarningIcon = (inventoryItem: InventoryItem) => {
+    // renders the red or yellow warning symbol if necesary
+    if (typeof inventoryItem.itemDefinition === 'string') {
+      return
+    }
+
+    if (
+      inventoryItem.quantity <
+      inventoryItem.itemDefinition.criticalStockThreshold
+    ) {
+      return <WarningIcon fontSize="small" sx={{ color: colors.warning.red }} />
+    }
+
+    if (
+      inventoryItem.quantity < inventoryItem.itemDefinition.lowStockThreshold
+    ) {
+      return (
+        <WarningIcon fontSize="small" sx={{ color: colors.warning.yellow }} />
+      )
+    }
+  }
+
+  const renderAttributeChips = (inventoryItem: InventoryItem) => {
+    return inventoryItem.attributes?.map((itemAttribute, i) => {
+      // attributes that are strings or numbers show the attribute name
+      // attributes that are list types do not
+
+      // TODO: remove this if statement after database types are split between
+      // response and request
+      if (typeof itemAttribute.attribute === 'string') {
+        return (
+          <Chip
+            size="small"
+            label={`${itemAttribute.attribute}: ${itemAttribute.value}`}
+            key={i}
+          />
+        )
+      }
+
+      if (typeof itemAttribute.attribute.possibleValues === 'object') {
+        return (
+          <Chip
+            size="small"
+            label={itemAttribute.value}
+            color="primary"
+            sx={{ backgroundColor: itemAttribute.attribute.color }}
```

Comment:
I played around with it, and this function will do the trick for now

```
function getContrastYIQ(hexcolor: string) {
    var colorNoHash = hexcolor.substring(1, hexcolor.length - 1)
    var r = parseInt(colorNoHash.substring(0, 2), 16)
    var g = parseInt(colorNoHash.substring(2, 2), 16)
    var b = parseInt(colorNoHash.substring(4, 2), 16)
    var yiq = (r * 299 + g * 587 + b * 114) / 1000
    return yiq >= 128 ? 'black' : 'white'
  }
```

Don't ask me the math behind it, but it measures the contrast and gives which text color would be better.
We may have to dive a bit deeper into color theory to improve this in the future, whether it be using a different calculation or not having a binary text color, but it will do for now.

Reply:
Add the following to target the chip text color:
'& .MuiChip-label': {color: [color])